purenix/src/Nix/Print.hs

```diff
@@ -121,7 +124,7 @@ binding (ident, body) = escape ident <> " = " <> body <> ";"
 
 escape :: Text -> Printer
 escape t =
-  if T.all isAlphaNum t && not (t `elem` nixKeywords)
+  if T.all (\c -> isAlphaNum c || c == '_') t && not (t `elem` nixKeywords)
```

Comment:
Without this additional check, the Nix file gets rendered like the following:

```nix
modules: 
let
  "__ffi" = 
{ mySubstring = builtins.substring; }
;
  mySubstring = __ffi.mySubstring;
  useMySubstring = modules.Main.mySubstring 0 3 "nixos";
  myId = a: a;
  myNum = modules.Main.myId modules.Dep.myDep;
in
  { inherit myId myNum mySubstring useMySubstring __ffi;
    inherit (modules.Dep) myDep;
  }
```

Reply:
...which does appear to actually work:

```console
$ nix eval --expr 'let "x" = 3; in x'
3
```

Well now I feel dumb.  I was almost certain that quoting let bindings didn't work in Nix.